### PR TITLE
[feat]: support governance

### PR DESCRIPTION
**Pull request by eigmax**

### CODE REVIEW

File: contracts/SafeMath.sol [hunk 1]
```diff
@@ -0,0 +1,206 @@
+pragma solidity ^0.8.0;
```
Comment (eigmax) [inline contracts/SafeMath.sol:1]: import this lib from openzepplin.

File: contracts/Timelock.sol [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    ) public {
+        require(msg.sender == admin, 'Timelock::cancelTransaction: Call must come from admin.');
+
+        bytes32 txHash = keccak256(abi.encode(target, value, signature, data, eta));
+        queuedTransactions[txHash] = false;
+
+        emit CancelTransaction(txHash, target, value, signature, data, eta);
+    }
+
+    function executeTransaction(
```
Comment (eigmax) [inline contracts/Timelock.sol:117]: reuse those functions from transactionModule

File: contracts/GovernanceToken.sol [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+    /**
+     * @notice Delegates votes from signatory to `delegatee`
+     * @param delegatee The address to delegate votes to
+     * @param nonce The contract state required to match the signature
+     * @param expiry The time at which to expire the signature
+     * @param v The recovery byte of the signature
+     * @param r Half of the ECDSA signature pair
+     * @param s Half of the ECDSA signature pair
+     */
+    function delegateBySig(
```
Comment (eigmax) [inline contracts/GovernanceToken.sol:370]: wallet.sol has the same implementation for delegateBySig. plz reuse it.

File: test/governance_token.test.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+describe('Governance Token', () => {
+    let governanceToken: Contract
+
+    let wallet
+    let other0
+    let other1
+    before(async () => {
+        ;[wallet, other0, other1] = await hre.ethers.getSigners()
```
Comment (eigmax) [inline test/governance_token.test.ts:26]: Remove the leading semicolon

**Review by JangoCCC — APPROVED** [review]

File: contracts/SafeMath.sol (hunk 1 above)
Comment (Weber-H2O2) [reply contracts/SafeMath.sol]: OK. Already done.

File: test/governance_token.test.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+const hre = require('hardhat')
+
+const helpers = require('./helpers')
+import GovernanceToken from '../artifacts/contracts/GovernanceToken.sol/GovernanceToken.json'
+
+chai.use(solidity)
+
+const provider = waffle.provider
+
+function expandTo18Decimals(n: number): BigNumber {
```
[the remark this answers is not in the document]
Comment (Weber-H2O2) [reply test/governance_token.test.ts]: OK. Already done.

**Pull request merged**